cep-releases.md

```diff
@@ -0,0 +1,72 @@
+<table>
+<tr><td> Title </td><td> Conda Release Schedule </td>
+<tr><td> Status </td><td> Draft </td></tr>
+<tr><td> Author(s) </td><td> Ken Odegard &lt;kodegard@anaconda.com&gt; </td></tr>
+<tr><td> Created </td><td> May 20, 2022 </td></tr>
+<tr><td> Updated </td><td> May 20, 2022 </td></tr>
+<tr><td> Discussion </td><td> NA </td></tr>
+<tr><td> Implementation </td><td> NA </td></tr>
+</table>
+
+## Abstract
+
+Describes a release cadence for all future `conda` versions starting with 5.0.0.
+
+## Specification
+
+We propose a monthly release schedule where the tagging/release occurs during the week (per ISO 8601, Monday through Sunday) of the 15th.
+
+In a nod to our many different kinds of users we propose a deprecation policy (to be defined in a later CEP) that allows for a slower adoption rate. Users on a slower schedule would be encouraged to update once a quarter.
+
+Both of these policies will be encapsulated in the version scheme:
+- the major version will be an annual counter,
+- the minor version will correlate with the quarter,
+- the micro/patch version will generally correlate with the month within the quarter but will also be incremented if a bug fix/security patch needs an early release, and
+- this scheme will start with the July 2022 release of 5.0.0.
```

Comment:
I like the general idea, but I'm wondering if we shouldn't just switch conda to using CalVer, which is essentially what's being proposed here. That would reduce any potential miscommunication caused by preserving something that looks like SemVer but really isn't. (E.g., whether conda 5.0.0 or 6.0.0 would be completely different from the current 4.x release series).

Reply:
Definitely an option as it would create a clear distinction between "old" and "new" conda.

Another thing to keep in mind with a more regular release cycle like this it's not infeasible for us to start planning out feature changes. Especially if we are seeking to allow a quarterly update schedule for slower-moving orgs. New features are added at the start of the quarter and the final release in the quarter is dedicated to cleanup/bugfixes.

Personally, I'd hesitate with switching entirely over to CalVer as it's generally a one-way path (the major version would do a significant one-way jump, 4 to 22 or 2022). It's hard to switch back again if that's desired for any reason.

An alternative would be to mirror Python's 2 → 3. We can do a one-time major version increment to 5 and drop the quarterly counter.

- 5.0.0: July 2022
- 5.0.1: August 2022 
- 5.0.2: September 2022
- ...
- 5.0.11: June 2022
- 5.1.0: July 2023
- 5.1.1: August 2023
- 5.1.2: September 2023
- ...

This is simpler but we'd lose the quarterly info but that could be instead solved with an LTS policy?

Reply:
Here are the current ideas and a new one. If we don't want to commit to a monthly release we should avoid a monthly counter as it'll result in versions jumping (e.g. 22.6 to 22.8 for bi-monthly releases) which could be confusing for users.

1. **YY.Q.MM**
    **\<year counter>.\<zero-based quarter counter>.\<zero-based month in quarter counter>**
    **<5+>.<0-3>.<0-3>**
    My original idea. Not in love with this idea as it made the version messy but nothing a little automation couldn't fix.
    - 5.0.0...5.0.1......5.1.0.........6.0.0
2. **YY.MM.MICRO**
    **\<year without century>.\<zero-based month counter>.\<zero-based release in month counter>**
    **<22+>.<0-11>.<0+>**
    @jezdez's proposed idea. This is _almost_ what we've already been using in https://github.com/conda/actions/releases. This would result in successful releases for a given month not having any micro/patch releases. Would also mean that the versions would jump if a release doesn't occur every month.
    - 22.0.0...22.0.1......22.1.0.........23.0.0
3. **YY.Q.MICRO**
    **\<year without century>.\<zero-based quarter counter>.\<zero-based release in quarter counter>**
    **<22+>.<0-3>.<0+>**
    This ends up looking very similar to @jezdez proposal but allows for slower releases, i.e. if we don't want to commit to monthly releases but want at least one per quarter.
    - 22.0.0...22.0.1......22.1.0.........23.0.0

Reply:
Ok adopted `YY.MM.MICRO` CalVer scheme on a bimonthly schedule.